daemon/fqdn.go

```diff
@@ -133,7 +132,10 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState, preCache
 		}}
 
 	d.dnsRuleGen = fqdn.NewRuleGen(cfg)
-	d.dnsPoller = fqdn.NewDNSPoller(cfg, d.dnsRuleGen)
+	pollerDNSCache := fqdn.NewDNSCacheWithLimit(
```

Comment:
Was there a reason to do this like this? instead of using poller.GetDNSCache() ? I guess you wanted to avoid passing variables to `NewDNSPoller`? We pass cfg in, and the perHostIPs limit and the minTTL should live there, so that's one place to pass them through. In fact, since we create cfg just above, this is a good place to handle setting the defaults for minTTL and for perHostIPs and simply pass that around.

Reply:
If we use the same poller as the general one, the DNS History will not work, because on GC we deleted, and after that we restore, in this case nothing to restore.

Reply:
Oh, I meant that we create `pollerDNSCache` here to use it later in the code, instead of using `d.poller.GetDNSCache()` when we do the GC. I would expect that the global DNSCache and the DNSCache (poller.DNSHistory) are still separate things.